scico/solver.py

```diff
@@ -229,25 +230,36 @@ def minimize(
         min_func = _wrap_func(func_, x0_shape, x0_dtype)
         jac = False
 
-    res = spopt.minimize(
-        min_func,
-        x0=x0,
-        args=args,
-        jac=jac,
-        method=method,
-        options=options,
+    res_dict = {}
```

Comment:
This might be cleaner using the following approach:
```python
    outside = None
    def f():
        nonlocal outside
        outside = 3
    print(outside)
    f()
    print(outside)
```

Reply:
On it!

Reply:
Worked just fine, thanks!

Reply:
MyPy is complaining about res not having an attribute when overriden with the actual OptimizeResult

should I do: 

res = spopt.OptimizeResult({'x':""})

or should I revert to using the dictionary approach for the side effect?